frontend/src/components/UserProfile/UserProfile.tsx

```diff
@@ -34,23 +34,61 @@ interface Props {
   numFollowing: number;
   numFollowers: number;
   fetchUserWithFollows: Function;
+  followUser: Function;
+  unfollowUser: Function;
 }
 
 const _UserProfile = ({
   user,
   numFollowing,
   numFollowers,
   fetchUserWithFollows,
+  followUser,
+  unfollowUser,
 }: Props) => {
   let { id } = useParams();
 
   const [cookies, setCookies] = useCookies();
+  const [loading, setLoading] = useState(false);
+  const [isFollowing, setIsFollowing] = useState(false);
+
+  const checkFollowing = () => {
+    let logged_in_user_id = cookies.user.id;
+
+    let followers = user?.followers || [];
+    let followers_id: number[] = followers?.map(
+      (follower) => follower['from_id']
+    );
+    setIsFollowing(followers_id.includes(logged_in_user_id));
+  };
+
+  const handleFollowClick = (e: any) => {
+    setLoading(true);
+    if (isFollowing) {
+      unfollowUser(user?.id, cookies.token, () => {
+        setLoading(false);
+        setIsFollowing(false);
+        fetchUserWithFollows(cookies.token, id);
+      });
+    } else {
+      followUser(user?.id, cookies.token, () => {
+        setLoading(false);
+        setIsFollowing(true);
+        fetchUserWithFollows(cookies.token, id);
+      });
+    }
```

Comment:
This looks redundant, please refactor. If you have problems regarding on this, please don't hesitate to ask me.

Reply:
done po